Fix async and some other stuff

HugoGranstrom opened the pull request:
This PR fixes #59 and #57 (partially). 

The compile options `-d:nimibParallelBuild=true/false` and `-d:nimibMaxProcesses=n` are introduced to control the parallel build of documents and the maximum number of concurrent builds to run at the same time.

HugoGranstrom commented:
Okay, this needs the parallel bug fix from nimib implemented [here](https://github.com/pietroppeter/nimib/pull/163).

HugoGranstrom commented:
It's green! When building Norm and Snorlogue locally I used this branch so it's basically already checked, but I'll go through them once more and double check. 

Anything else you want done in this PR?

pietroppeter commented:
The only thing I can think of is whether or not we mentioned somewhere that you could not use nbCode in some templates. If we did (I could not find a mention but used only GitHub search) maybe we should amend it

pietroppeter commented:
Haha, I thought I was commenting on the other PR on nimib nbCode...

pietroppeter commented:
quoting from the updated readme's contribute section (wip in #56) ;)

> Once you make a change, remember to document your changes somewhere appropriate
> and to update the [changelog](changelog.md)
> (and if needed also the [src/readme.md](src/readme.md))! 

for the moment just update the changelog, for documentation I will do it in the other PR (the only thing we need to mention is the switch for turning off parallel builds

HugoGranstrom commented:
> The only thing I can think of is whether or not we mentioned somewhere that you could not use nbCode in some templates. If we did (I could not find a mention but used only GitHub search) maybe we should amend it

I added an example that doesn't work in https://github.com/pietroppeter/nimib/issues/134 . Maybe should add it to changelog as well?

> for the moment just update the changelog, for documentation I will do it in the other PR (the only thing we need to mention is the switch for turning off parallel builds

The changelog that doesn't exist anywhere except in #56 :scream: I could try to rebase this PR on top of #56, oooorrrrr I could write the changelog here and ask you kindly to include it in #56 :rofl: :see_no_evil: 

```
* Parallel build fixes
  * Fixed parallel build so it uses the correct folder for saving log files.
  * The error messages are printed to terminal as well
  * Pass compiler flag `-d:parallelBuild=false` to compile in serial.
* The next-arrow is shown correctly on the second to last page.
```

pietroppeter commented:
Ah no, it is fine here in the issue, I will add them in the change log

pietroppeter commented:
> I added an example that doesn't work in pietroppeter/nimib#134 . Maybe should add it to changelog as well?

Yeah that could be a good idea. With the old system it would have worked that example?

HugoGranstrom commented:
Thanks :D 

> Yeah that could be a good idea. With the old system it would have worked that example?

Will add it then :+1: In the old system I think it would have given in the best case undefined behavior and in the worst case an index error (the endPos is calculated the same way as before and it would have given us the wrong file). So now we have actually just started handling this example as an error instead of ignoring it.

HugoGranstrom commented:
Something else I've noticed, especially when scinim is that it lags my laptop to run all of them at once. They are fairly heavy computations happening in some those tutorials. So what if we have 100 such tutorials? Running them all at once wouldn't be great. So I've added a `-d:nimibMaxProcesses` so that we have a maximum amount of build running in parallel. I have updated my changelog above.

With this I'm pretty satisfied with this PR.
